src/app/layout.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react'
+/* import Router from 'next/router' */
+import type {Metadata} from 'next'
+import Script from 'next/script'
+import {Preconnect} from './preconnect'
+import {pageview, GA} from '../lib/gtag'
+import siteMeta from '../config/siteMeta'
+
+import '../style/style.css'
+import '../style/prism-plain.css'
+
+/* import Router from 'next/router' */
+/* Router.events.on('routeChangeComplete', (url) => pageview(url)) */
+
+const {
+	description,
+	social: {twitter, github, linkedin, mastodon},
+	twitterId,
+	siteUrl,
+	author,
+	title,
+} = siteMeta
+
+const fullTitle = `${author} | ${title}`
+
+export const metadata: Metadata = {
+	metadataBase: new URL(siteUrl),
+	title: {
+		template: `${fullTitle} - %s`,
+		default: fullTitle,
+	},
+	description,
+	openGraph: {
+		title: fullTitle,
+		description,
+		url: siteUrl,
+		images: '/img/fb-image.jpg',
+		type: 'website',
+		siteName: author,
+		// @NOTE move to config
+		locale: 'en-US',
+	},
+	alternates: {
+		canonical: siteUrl,
+	},
+	// @TODO
+	// keywords: ['Next.js', 'React', 'JavaScript'],
+	authors: [{name: author, url: siteUrl}],
+	creator: author,
+	publisher: author,
+	themeColor: '#1f2325',
+	colorScheme: 'dark light',
+	appleWebApp: {
+		title: siteUrl,
+		statusBarStyle: 'black-translucent',
+	},
+	twitter: {
+		// @ts-ignore
+		card: 'summary_large_image',
+		title: fullTitle,
+		description,
+		siteId: twitterId,
+		creator: twitter.display,
+		creatorId: twitterId,
+		images: ['/i/social-image.jpg'],
+		domain: siteUrl,
+	},
+	viewport: {
+		width: 'device-width',
+		initialScale: 1,
+		maximumScale: 1,
+		viewportFit: 'cover',
+	},
+	other: {
+		'application-name': siteUrl,
+	},
+}
+
+export default function RootLayout({
+	// Layouts must accept a children prop.
+	// This will be populated with nested layouts or pages
+	children,
+}: {
+	children: React.ReactNode
+}) {
+	return (
+		<html lang="en" className="text-xl">
+			<head>
+				<Preconnect />
+				{[twitter, github, linkedin, mastodon].map(({url}) => (
+					<link key={url} href={url} rel="me" />
+				))}
+				<link
+					href="https://fonts.googleapis.com/css2?family=Playfair+Display:ital,wght@0,400;0,700;1,400;1,700&display=swap"
+					rel="stylesheet"
+				/>
+				<GA />
```

Comment:
Check this https://github.com/ahmedelgabri/imantohami.com/pull/17/files#diff-29d0a55f5ca1a9951042587f8b891bb4cd8b204f5bc855dc2468ed44cde2f5b3R94